lib/worker.js

```diff
@@ -20,12 +20,21 @@ const RateLimiterNoCluster = require('./ratelimiter').nocluster;
  * @constructor
  */
 class Worker extends BaseService {
+    constructor() {
+        super();
+
+        this.shutdownHandler = null;
+        this.dumpHeapHandler = null;
+        this.messageHandler = null;
+        this.serviceStatusHandler = null;
+    }
+
     _getConfigUpdateAction() {
         return new P((resolve, reject) => {
             const timeout = setTimeout(() => {
                 reject(new Error(`Timeout waiting for config in worker ${process.pid}`));
             }, 3000);
-            process.on('message', (message) => {
+            process.on('message', this.messageHandler = function (message) {
```

Comment:
Why chaging it to `function`? It will create a new scope for `this` and screw line 40

Reply:
idem

Reply:
Changed to arrow function